pathway-analysis/ssgsea_example.Rmd

```diff
@@ -0,0 +1,392 @@
+([Barbie et al. _Nature_. 2009.](https://dx.doi.org/10.1038/nature08460)),
+which is implemented in the 
+[`GSVA` bioconductor package](https://bioconductor.org/packages/release/bioc/html/GSVA.html).
+Note that `GSVA` contains _multiple_ gene set enrichment methods and has an 
+excellent [vignette](https://bioconductor.org/packages/release/bioc/vignettes/GSVA/inst/doc/GSVA.pdf).
+
+## Gene sets
+
+We will use KEGG (Kyoto Encyclopedia of Genes and Genomes) Pathways in this
+analysis.
+We acquired the KEGG `v6.2` pathway set that used human Entrez IDs from the 
+[Molecular Signatures Database (MSigDB)](http://software.broadinstitute.org/gsea/msigdb).
+
+In the [`kegg_ortholog_mapping`](./kegg_ortholog_mapping.nb.html) notebook in 
+this module, we mapped human Entrez IDs to mouse symbols using the 
+[`hcop` package](https://github.com/stephenturner/hcop).
+When there was a 1:many mapping between human Entrez IDs and mouse gene symbols,
+we selected the mouse gene symbol with the highest number of resources 
+supporting the mapping.
+This decision might not be suitable for every experiment and may result in
+some loss of information.
+
+## Dataset
+
+We're using [`GSE75574`](https://www.refine.bio/experiments/GSE75574/gene-expression-in-mouse-tissues-in-response-to-short-term-calorie-restriction) 
+in this notebook. 
+This dataset assays the gene expression response to short-term calorie
+restriction in multiple tissues from multiple mouse strains.
+This is the same dataset we used in one of our QuSAGE examples 
+([`qusage_single_dataset`](./qusage_single_dataset.nb.html)), where we only 
+looked at white adipose tissue.
+
+## Set up
+
+Package installation and loading.
+
+```{r}
+if (!("qusage" %in% installed.packages())) {
+  BiocManager::install("qusage", update = FALSE)
+}
+
+if (!("org.Mm.eg.db" %in% installed.packages())) {
+  BiocManager::install("org.Mm.eg.db", update = FALSE)
+}
+
+if (!("GSVA" %in% installed.packages())) {
+  BiocManager::install("GSVA", update = FALSE)
+}
+
+if(!("matrixStats" %in% installed.packages())) {
+  install.packages("matrixStats")
+}
+```
+
+```{r}
+`%>%` <- dplyr::`%>%`
+```
+
+```{r}
+library(org.Mm.eg.db)
+library(GSVA)
+```
+
+Create directories to hold plots and results if they do not yet exist.
+
+```{r}
+if (!dir.exists("plots")) {
+  dir.create("plots")
+}
+
+if (!dir.exists("results")) {
+  dir.create("results")
+}
+```
+
+## Read in refine.bio data
+
+The gene expression matrix of the dataset we'll be working with is too large 
+to be tracked with git without compression, so we need to unzip it if we have
+not already.
+This compressed folder contains both the expression matrix and the metadata.
+
+```{r}
+# if the unzipped folder does not exist
+if (!dir.exists(file.path("data", "GSE75574"))) {
+  compressed_file <- file.path("data", "GSE75574.zip")
+  unzip(compressed_file, exdir = "data")
+}
+```
+
+### Expression data
+
+Because we used this in another notebook, it's possible that you already have
```

Comment:
Should you also add this same `if` statement thing in the other notebook where this dataset is used, in case someone uses this notebook first?

Reply:
I expect, with where I think the README for this module is headed/will be organized, that folks will go through the other notebook first. Leaving that notebook as is, broken into multiple chunks, leaves more room for exposition.